apache/pinot
# Expose metric for table rebalance

This PR adds a metric to track table rebalance. This metric can be useful in cases wherein we don't want certain alerts (like ingestion stopped) to trigger as a table rebalance operation is going on.

Testing steps:
- Start `RealtimeQuickStart.java` with 3 servers.
- Comment out `if(segmentAssignmentUnchanged)` block in `TableRebalancer.java` so that the dataflow actually hits the table rebalance observer.
- Rebalance the table and checkout `localhost:9021/metrics`.
- You'll see the metric (exported using the catch-all regex).

Here's the output:
 
<img width="1728" alt="Screenshot 2024-01-17 at 10 32 18 AM" src="https://github.com/apache/pinot/assets/84911643/9a9b9ab5-1f08-4fab-a48d-8d1516b8c3f7">

Also, for table paused and ingestion paused, we use the catch-all regex now. Test results:

<img width="973" alt="Screenshot 2024-01-17 at 5 17 51 PM" src="https://github.com/apache/pinot/assets/84911643/08682762-a2c6-4334-ad46-9eaca8ef311a">

<img width="795" alt="Screenshot 2024-01-17 at 5 18 05 PM" src="https://github.com/apache/pinot/assets/84911643/09e1c15f-eea5-4240-b758-345b22685693">
